synapse/cores/common.py

```diff
@@ -1938,3 +1912,68 @@ def _tufosByInetCidr(self, prop, valu, limit=None):
 
         return self.getTufosBy('range', prop, (ipv4addr, ipv4addr+mask), limit=limit)
 
+    #############################################################
+    # support typelib persistence
+    #############################################################
+
+    # overrides: synapse.lib.types.TypeLib.addType
+    def addType(self, typ):
+        s_types.TypeLib.addType(self, typ)
+        self.formTufoByFrob('syn:type', typ.name)
+
+    # overrides: synapse.lib.types.TypeLib.addSubType
+    def addSubType(self, name, subof, **info):
+        s_types.TypeLib.addSubType(self, name, subof, **info)
+        base = self.getTufoByFrob('syn:type', subof)
+
+        subinfo = {}
+        for propname, propval in base[1].items():
+            if not propname.startswith('syn:type:'):
+                continue
+            propname = propname.partition('syn:type')[2]
```

Comment:
It gets more confusing once you've started using namespaces to differentiate my:host:prop versus their:host:prop. Why are you overriding the datamodel functions instead of modifying them?